persistence/jdbc-common/src/main/java/org/infinispan/persistence/jdbc/common/impl/connectionfactory/CDIConnectionFactory.java

```diff
@@ -0,0 +1,93 @@
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+import org.infinispan.commons.util.ReflectionUtil;
+import org.infinispan.commons.util.Util;
+import org.infinispan.persistence.jdbc.common.configuration.CDIConnectionFactoryConfiguration;
+import org.infinispan.persistence.jdbc.common.configuration.ConnectionFactoryConfiguration;
+import org.infinispan.persistence.jdbc.common.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.jdbc.common.logging.Log;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.util.logging.LogFactory;
+
+import jakarta.enterprise.inject.spi.CDI;
+
+/**
+ * Connection factory that can be used when on CDI environments.
+ */
+public class CDIConnectionFactory extends ConnectionFactory {
+
+   private static final Log log = LogFactory.getLog(CDIConnectionFactory.class, Log.class);
+
+   private CDIConnectionFactoryConfiguration managedConfiguration;
+   private volatile DataSource dataSource;
+
+
+   @Override
+   public void start(ConnectionFactoryConfiguration factoryConfiguration, ClassLoader classLoader) throws PersistenceException {
+      if (factoryConfiguration instanceof CDIConnectionFactoryConfiguration) {
+         managedConfiguration = (CDIConnectionFactoryConfiguration) factoryConfiguration;
+      } else {
+         throw new PersistenceException("FactoryConfiguration has to be an instance of CDIConnectionFactoryConfiguration");
+      }
+   }
+
+   private void initDataSource() {
+      try {
+         String datasourceName = managedConfiguration.name();
+         if (datasourceName == null) {
+            dataSource = CDI.current().select(DataSource.class).get();
+         } else {
+            Class<? extends Annotation> annotationClass = (Class<? extends Annotation>) ReflectionUtil.getClassForName(managedConfiguration.annotation(), Thread.currentThread().getContextClassLoader());
+            Annotation annotation = Util.newInstanceOrNull(annotationClass, new Class[]{String.class}, datasourceName);
+            dataSource = CDI.current().select(DataSource.class, annotation).get();
+         }
+         if (dataSource == null) {
+            PERSISTENCE.connectionNotFound("CDI", datasourceName);
+            throw new PersistenceException(String.format("Could not find a connection in jndi under the name '%s'", datasourceName));
+         }
+      } catch (ClassNotFoundException e) {
+         throw new RuntimeException(e);
+      }
+   }
+
+   @Override
+   public void stop() {
+   }
+
+   @Override
+   public Connection getConnection() throws PersistenceException {
+      if (dataSource == null)
+         initDataSource();
+
+      Connection connection;
+      try {
+         connection = dataSource.getConnection();
+      } catch (SQLException e) {
+         PERSISTENCE.sqlFailureRetrievingConnection(e);
+         throw new PersistenceException("This might be related to https://jira.jboss.org/browse/ISPN-604", e);
```

Comment:
link isn't working for me
```suggestion
         throw new PersistenceException("This might be related to https://issues.redhat.com/browse/ISPN-604", e);
```

Reply:
Does it make sense to even include this Jira reference given how few of our users actually use XATransactions?